File: site_web_arosaje_ball/src/models/Plante.ts
import StatutPlante from "@/models/StatutPlante";
import Personne from "@/models/Personne";
import BibliothequePlante from "@/models/BibliothequePlante";
import PhotoPlante from "@/models/PhotoPlante";

export default class Plante {

  private id: number;
  private localisation: string;
  private bibliothequePlante: BibliothequePlante;
  private proprietaire: Personne;
  private photo?: PhotoPlante;


  constructor(id: number, localisation: string, bibliothequePlante: BibliothequePlante, proprietaire: Personne, photo?: PhotoPlante) {
    this.id = id;
    this.localisation = localisation;
    this.bibliothequePlante = bibliothequePlante;
    this.proprietaire = proprietaire;
    this.photo = photo;
  }


  getId(): number {
    return this.id;
  }

  setId(value: number) {
    this.id = value;
  }


 getLocalisation(): string {
    return this.localisation;
  }

  setLocalisation(value: string) {
    this.localisation = value;
  }

 getBibliothequePlante(): BibliothequePlante {
    return this.bibliothequePlante;
  }

  setBibliothequePlante(value: BibliothequePlante) {
    this.bibliothequePlante = value;
  }

  getProprietaire(): Personne {
    return this.proprietaire;
  }

  setProprietaire(value: Personne) {
    this.proprietaire = value;
  }

  getPhoto(): PhotoPlante {
    return <PhotoPlante>this.photo;
  }

  setPhoto(value: PhotoPlante) {
    this.photo = value;
  }
}

File: front-end/lib/src/models/Plante.dart
import 'dart:convert';

import 'package:arosaje/src/models/BibliothequePlante.dart';
import 'package:arosaje/src/models/GuidePlante.dart';
import 'package:arosaje/src/models/Personne.dart';
import 'package:arosaje/src/models/StatutPlante.dart';
import 'package:arosaje/src/services/photoPlanteService.dart';
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

import 'PhotoPlante.dart';

class Plante {
  final int id;
  final String localisation;
  final BibliothequePlante bibliothequePlante;
  final Personne proprietaire;
  final StatutPlante statut;

  const Plante({
    required this.id,
    required this.localisation,
    required this.bibliothequePlante,
    required this.proprietaire,
    required this.statut,
  });

  static ListView getListPlantes(List<Plante> plantes)
  {
    List<Widget> list = [];
    for(var i = 0; i < plantes.length; i++){
      var photo = getUrlOfPhotoPlante(plantes[i].id);

      list.add(Card(
        child: Column(
          children: <Widget> [
           // Image.network(photo.),
            ListTile(
              title: Text(plantes[i].bibliothequePlante.nom),
              subtitle: Text(plantes[i].bibliothequePlante.typePlante.description),
            ),
          ]
        ),
      ));
    }
    return ListView(children: list);
  }
  static List<Plante> listFromJson(List<dynamic> parsedJson) {

    List<Plante> plantes = <Plante>[];
    for (var plante in  parsedJson) {
        plantes.add(Plante.fromJson(plante));
    }
    return plantes;
  }

  factory Plante.fromJson(Map<String, dynamic> json) {
    return Plante(
      id: json['id'],
      localisation: json['localisation'],
      bibliothequePlante:  BibliothequePlante.fromJson(json['bibliothequePlante']),
      proprietaire: Personne.fromJson(json['proprietaire']),
      statut: StatutPlante.fromJson(json['statut']),
    );
  }
}
